src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -2369,6 +2369,26 @@ size_t MergeTreeData::getMaxPartsCountForPartition() const
         res = std::max(res, cur_count);
     }
 
+    if (inactive)
```

Comment:
The interface of this function is somewhat dirty.
Maybe simply add a separate function `getMaxInactivePartsCountForPartition`?